UI layout for About and How it Works sections

lukemckinstry opened the pull request:
### Summary
This PR implements the designs for the "About" & "How it Works" sections of the UI. All changes are to the `index.html` file, no changes were needed to the configs in `tailwind.config.js` or `input.css`.

Adds images and text from designs. 

### Implements
 Issue #16

### Demo
mobile (`<sm`)
![demo-rvw2-issue16-mobile](https://user-images.githubusercontent.com/16727618/204553316-33c307a8-4dc6-4cc5-9ab0-139d1c9090ff.png)

`sm`
![demo-rvw2-issue16-sm](https://user-images.githubusercontent.com/16727618/204553817-7356ad7d-037d-4136-a4d7-033457323b21.png)

`>=md`
![demo-rvw2-issue16-md](https://user-images.githubusercontent.com/16727618/204553980-97504bc3-d031-45f8-bdde-174e5bbfbfbd.png)

### Testing
 - visually inspect layout at various screen sizes and orientations
 - run linter when one has been selected for the project

ktohalloran reviewed (changes requested):
This looks great! Just a few small fix ups:
- I know I said there was a nice mobile friendly skeleton already in place, but unfortunately I missed a few things and this would be the best time to address them.
     - The "Build intelligent geospatial applications with deep learning." title should shrink on mobile screens to the `text-2xl` font size.
     - The color of the font of the "How it Works" and "Example is showing a semantic segmentation task" headlines should be a gray color; unfortunately, this is not one of those colors that I built in at the start so you'll need to add it to the `textColor` list in `tailwind.config.js`.
- The image in the "About" section doesn't appear centered vertically, especially on mobile screens. I know that it is actually centered, but since the shadow below it is adding to its height, it appears off-centered. Can you try out a few ways of getting it closer to visual center? This isn't a huge deal, though, so if making it responsive and visually centered is A Thing, we can punt it for now.
- We should add the links to the linked buttons now; while I know that the "Explore documentation" one should link to "https://docs.rastervision.io/", I need to verify where the "Start training with GroundWork" one should go. @gatesgodin do you know the answer to this question?

lukemckinstry commented:
Added a commit with the following style fixups:
 * Changed "Build intelligent geospatial applications with deep learning." to `<h2>` 
 * Added `lightGray` and used it for "How it Works" and "Example is showing a semantic segmentation task" headlines 
 * Added 40px of top padding to the image in the "About" section to make it appear vertically centered.
 * Added "https://docs.rastervision.io/" to the "Explore Documentation" button, still waiting for url to add to "Start training with GroundWork"

ktohalloran commented on src/index.html:
```diff
@@ -125,24 +130,70 @@ <h1 class="text-white mb-6">
+                    easy to read, reuse, and maintain.
+                  </p>
+                </div>
+                <a href="https://docs.rastervision.io/" target="_blank">
```
When I added links to the hero buttons, I decided not to have them open in a new tab, mostly because I figured the point of a marketing site was to get users to the product and probably wasn't helpful to have open alongside it. I'm not entirely convinced that that's the best play, but for now I think that the buttons should all do the same thing. 

```suggestion
                <a href="https://docs.rastervision.io/">
```

ktohalloran commented on src/index.html:
```diff
@@ -104,19 +104,24 @@ <h4 class="opacity-60">GitHub Discussions</h4>
           <div class="sm:flex sm:flex-row gap-10">
             <div class="basis-1/2 my-auto">
-              <h1 class="text-white mb-6">
+              <h2 class="text-white mb-6">
```
This should only shrink on mobile and small screens. I initially only said mobile in my first review but after implementing it, it looks better as a smaller font size on small screens, too.

```suggestion
              <h1 class="text-2xl text-white mb-6 md:text-3xl">
```

ktohalloran commented on src/index.html:
```diff
@@ -125,24 +130,70 @@ <h1 class="text-white mb-6">
     <section>
       <div class="bg-background-off-white">
         <div class="max-width-container text-center">
```
This is a bit close to the "About" section here on all screens, so we should add some space.

```suggestion
        <div class="max-width-container text-center mt-10">
```

ktohalloran commented on src/index.html:
```diff
@@ -104,19 +104,24 @@ <h4 class="opacity-60">GitHub Discussions</h4>
         <div class="max-width-container">
           <div class="sm:flex sm:flex-row gap-10">
```
This gets a little close to the hero component on mobile, so we should add a bit of space.

```suggestion
          <div class="mt-14 sm:flex sm:flex-row sm:mt-0 gap-10">
```

ktohalloran commented on src/index.html:
```diff
@@ -125,24 +130,70 @@ <h1 class="text-white mb-6">
+                    labeled.
+                  </p>
+                </div>
+                <button class="secondary-btn">
+                  Start training with GroundWork
+                </button>
```
```suggestion
                <button class="secondary-btn hover:bg-gray-100">
                  Start training with GroundWork
                </button>
```

I realized I never told you about the design system, which illustrates the hover state for the buttons. This implements that hover state for this type of button, though you'll have to do the same for the "Explore documentation" button.

Also, after thinking about it, I think this links to the same place as the "Get Started" hero button, so https://github.com/azavea/raster-vision. So go ahead and add that link now; I can fix it later if that's not right.

ktohalloran reviewed (changes requested):
This looks great! The centering of the "About" image looks much better, thanks for fixing that! I also think you did a great job implementing the responsiveness of the "How it Works" sections. Just a few more tiny fixups and I think you can also clean up the commit history a bit and squash all the commits down into one. But then it should be good to go after that!

lukemckinstry commented:
Added those additional style fixups. I didn't refactor to 1 commit yet but I can use squash and merge when the PR is approved.

ktohalloran reviewed (approved):
Looks great! Thanks for making all those changes! 👍

gatesgodin commented on src/index.html:
```diff
@@ -109,14 +109,14 @@ <h1 class="text-white mb-6">
+                Raster Vision is an open source library and framework that bridges 
+                the divide between the world of GIS and deep learning-based computer 
+                vision. It provides a configurable computer vision pipeline that works 
+                on <b>chip classification, semantic segmentation,</b> and <b>object detection.</b>
```
@ktohalloran It should go to the groundwork website. https://groundwork.azavea.com/
